src/codegen/llvm.zig

```diff
@@ -5909,7 +5902,15 @@ pub const FuncGen = struct {
         });
         defer assert(self.blocks.remove(inst));
 
-        try self.genBody(body);
+        if (o.di_builder) |dib| {
+            const old_scope = self.di_scope.?;
+            const lexical_block = dib.createLexicalBlock(old_scope, self.di_file.?, self.prev_dbg_line, self.prev_dbg_column);
+            self.di_scope = lexical_block.toScope();
+            try self.genBody(body);
+            self.di_scope = old_scope;
+        } else {
+            try self.genBody(body);
+        }
```

Comment:
Are trys, loops, switches and condbrs handled correctly? They don't seem to have been updated in any way.

Reply:
The scoping is currently just based on actual `block` instructions - this is okay because `AstGen` today emits blocks when it doesn't necessarily need to (e.g. `if (x) { ... } else { ... }` will put `block`s in both the condbr bodies). A future enhancement to handle this for all AIR bodies, alongside elision of unnecessary blocks at the `AstGen` level, would definitely be good.